modules/core/shared/src/main/scala/util/Typer.scala

```diff
@@ -177,7 +177,10 @@ object Typer {
   case class TypeInfo(oid: Int, name: String, arrayTypeOid: Option[Int], relOid: Option[Int])
 
   implicit class ProtocolOps[F[_]: Functor](p: Protocol[F]) {
-    val oid: Codec[Int] = Codec.simple(_.toString, _.toInt.asRight, Type.oid)
+    //Note Postgres defines oids as *unsigned* Ints https://www.postgresql.org/docs/current/datatype-oid.html
+    //Since Scala currently lacks a built-in unsigned Int type, if the oid exceeds `Int.MaxValue`
+    //it will be converted to/from a negative Int by this Codec (only observed in CockroachDB)
+    val oid: Codec[Int] = Codec.simple(java.lang.Integer.toUnsignedLong(_).toString, _.toLong.toInt.asRight, Type.oid)
```

Comment:
I accidentally directly pushed a slight revision of the parsing logic in e66ae4fe87d92d49cc8b5b4d4ebd2365bcc65301.